addons/hr_holidays_attendance/models/hr_leave.py

```diff
@@ -42,7 +42,7 @@ def write(self, vals):
             employee = leave.employee_id
             duration = leave.number_of_hours_display
             overtime_duration = leave.overtime_id.sudo().duration
-            if overtime_duration != duration:
+            if overtime_duration != -1 * duration:
```

Comment:
Hi @hwangh95, your fix is correct, but could you provide me with a flow where going in this condition is problematic functionally ?

Reply:
We're writing a customization that recomputes leaves based on the resource_calendar on a contract but we found that it was running into issues here when the duration was the same

Reply:
but if the employee has enough total_overtime it shouldn't be an issue no ? or is the objective here to allow negative overtime

Reply:
@jugj-odoo I think the exact use case we have is a bit hard to explain, but as I was coming up with an explanation I found a common use case it will fail on, even testable on runbot.

Let's say an employee has 2 hours overtime saved up. If she spontaneously wants to go home early and creates an overtime leave for 2 hours near the end of her day, her overtime is used up. Now if she goes on a lunch break, her overtime will be set to negative.

If her time off officer gets to the leave request after the lunch break and approves the leave request, it won't be possible

Reply:
@hwangh95 It's true that this case will fail, but then the case where an employee doesn't do her full hours on the next day gets bypassed as well, which should not be the case.

Reply:
Can you explain why that's the case? The overtime gets allocated already when the time off is created no? As long as the number_of_hours_display used is the same as the duration, the total_overtime shouldn't matter.

Reply:
@hwangh95 hmm, yeah now that i think about it we should look at the overtime at the moment the leave was taken